src/core/dom/query.js

```diff
@@ -331,3 +331,64 @@ export function elementByTag(element, tagName) {
   assertIsName(tagName);
   return element./*OK*/ querySelector(tagName);
 }
+
+/**
+ * Returns the original nodes of the custom element without any service
+ * nodes that could have been added for markup. These nodes can include
+ * Text, Comment and other child nodes.
+ *
+ * @param {!Node} element
+ * @return {!Array<!Node>}
+ */
+export function getRealChildNodes(element) {
+  return childNodes(element, (node) => !isInternalOrServiceNode(node));
+}
+
+/**
+ * Returns the original children of the custom element without any service
+ * nodes that could have been added for markup.
+ *
+ * @param {!Element} element
+ * @return {!Array<!Element>}
+ * @package @final
+ */
+export function getRealChildren(element) {
+  return childElements(element, (element) => !isInternalOrServiceNode(element));
+}
+
+/**
+ * Returns "true" for internal AMP nodes or for placeholder elements.
+ * @param {!Node} node
+ * @return {boolean}
+ */
+export function isInternalOrServiceNode(node) {
```

Comment:
AFAICT this isn't consumed by anything else
```suggestion
function isInternalOrServiceNode(node) {
```

Reply:
it is used by `custom-element` unfortunately.